auth/jwt/jwt.go

```diff
@@ -54,169 +52,130 @@ func NewVerifier(opts VerifierOptions) Verifier {
 	if opts.MaxCachedEntries == 0 {
 		opts.MaxCachedEntries = defaultMaxCachedEntries
 	}
+
 	return &verifier{
-		providers: opts.Providers,
-		cache:     cache.NewLRU(opts.CacheTTL, opts.CacheEvictionInterval, int32(opts.MaxCachedEntries)),
-		knownBad:  cache.NewLRU(defaultBadEntryCacheTTL, opts.CacheEvictionInterval, 100),
+		jwksManager: &jwksManager{
+			providers: opts.Providers,
```

Comment:
Not sure why would this be necessary? I'm already removing dups and calculating a minimum refresh time so that we don't make duplicate calls to any given provider URL...